llama-index-core/llama_index/core/memory/simple_composable_memory.py

```diff
@@ -81,13 +81,16 @@ def _compose_message_histories(
         """Get chat history."""
         # get from primary
         messages = self.primary_memory.get(input=input, **kwargs)
+        messages_text = " ".join(m.content or "" for m in messages)
 
         # get from secondary
         # TODO: remove any repeated messages in secondary and primary memory
         secondary_histories = []
         for mem in self.secondary_memory_sources:
             secondary_history = mem.get(input, **kwargs)
-            if len(secondary_history) > 0:
+            secondary_text = " ".join(m.content or "" for m in secondary_history)
+
+            if len(secondary_history) > 0 and secondary_text not in messages_text:
```

Comment:
Instead of checking if the entire strings match, should we instead be comparing each individual message?

With this, the vector memory could still return a partial overlap

Reply:
just sharing my message from slack:

> ....the logic for checking for duplicates is not always straightforward tho...what I got there so far is checking if the entire history for a secondary source,  represented as joined text is contained in the similar text for the primary history. i think this can be enhanced perhaps by using fuzzy logic for example... or computing a similarity score on these texts instead?

We can check individual message, I was concerned about removing a USER message but perhaps not the ASSISTANT message that follows it (probably an edge case...not sure how likely)